docs/single-page-app.md

```diff
@@ -0,0 +1,230 @@
+
+## Configuration reference
+
+### Path
+
+```yaml
+constructs:
+    landing:
+        type: single-page-app
+        path: public
+```
+
+The `path` option should point to the local directory containing the static website. Use `path: .` to upload the content of the current directory.
+
+All files in that directory will be deployed and made available publicly.
+
+When using a JavaScript bundler (for example when working with Webpack, VueJS, React, etc.), upload the compiled files. For example this could be the `dist/` directory.
+
+### Custom domain
+
+```yaml
+constructs:
+    landing:
+        # ...
+        domain: mywebsite.com
+        # ARN of an ACM certificate for the domain, registered in us-east-1
+        certificate: arn:aws:acm:us-east-1:123456615250:certificate/0a28e63d-d3a9-4578-9f8b-14347bfe8123
+```
+
+The configuration above will activate the custom domain `mywebsite.com` on CloudFront, using the provided HTTPS certificate.
+
+After running `serverless deploy` (or `serverless info`), you should see the following output in the terminal:
+
+```
+landing:
+  url: https://mywebsite.com
+  cname: s13hocjp.cloudfront.net
+```
+
+Create a CNAME DNS entry that points your domain to the `xxx.cloudfront.net` domain. After a few minutes/hours, the domain should be available.
+
+#### HTTPS certificate
+
+To create the HTTPS certificate:
+
+- Open [the ACM Console](https://console.aws.amazon.com/acm/home?region=us-east-1#/wizard/) in the `us-east-1` region (CDN certificates _must be_ in us-east-1, regardless of where your application is hosted)
+- Click "_Request a new certificate_", add your domain name and click "Next"
+- Choose a domain validation method:
+  - Domain validation will require you to add CNAME entries to your DNS configuration
+  - Email validation will require you to click a link in an email sent to `admin@your-domain.com`
+
+After the certificate is created and validated, you should see the ARN of the certificate.
+
+#### Multiple domains
+
+It is possible to set up multiple domains:
+
+```yaml
+constructs:
+    landing:
+        # ...
+        domain:
+            - mywebsite.com
+            - app.mywebsite.com
+```
+
+### Error page
+
+By default, all 404 requests are redirected to `index.html` with a 200 response status. This behavior is optimized for Single-Page Applications: it allows doing client-side URL routing with JavaScript frameworks.
+
+For static websites that _are not SPA_, it is possible to serve a custom "Not found" error page:
+
+```yaml
+constructs:
+    landing:
+        # ...
+        errorPage: error.html # can be any HTML file in your project
+```
+
+When a browser requests the URL of a non-existing file, the `error.html` file will be served with a 404 response status.
+
+Do not use this setting when doing JavaScript URL routing: this will break URL routing.
```

Comment:
Should we remove the `errorPage` option from the single-page app construct? Any reason to keep it?

Reply:
There are no reason to keep it. I will remove it :wink:

Reply:
Done